indexer/editable_map_object.cpp

```diff
@@ -49,14 +95,49 @@ vector<feature::Metadata::EType> const & EditableMapObject::GetEditableFields()
 
 StringUtf8Multilang const & EditableMapObject::GetName() const { return m_name; }
 
-vector<LocalizedName> EditableMapObject::GetLocalizedNames() const
+NamesDataSource EditableMapObject::GetNamesDataSource() const
 {
-  vector<LocalizedName> result;
-  m_name.ForEach([&result](int8_t code, string const & name) -> bool
-                 {
-                   result.push_back({code, name});
-                   return true;
-                 });
+  return GetNamesDataSource(m_name, GetMwmLanguages(), languages::GetCurrentNorm());
+}
+
+// static
+NamesDataSource EditableMapObject::GetNamesDataSource(StringUtf8Multilang const & source,
+                                                      vector<string> const & nativeMwmLanguages,
+                                                      string const & userLanguage)
+{
+  NamesDataSource result;
+  vector<LocalizedName> & names = result.names;
+  size_t & mandatoryCount = result.mandatoryNamesCount;
+  // Push Mwm languages
+  mandatoryCount = PushMwmLanguages(source, nativeMwmLanguages, names);
+
+  // Push user's language
+  int8_t const index = StringUtf8Multilang::GetLangIndex(userLanguage);
+  if (PushLanguage(source, index, names))
+    ++mandatoryCount;
+  
+  // Push international language
```

Comment:
Dot in the end, please. See https://github.com/mapsme/omim/blob/master/docs/CPP_STYLE.md